Assignment6/Q1.py

```diff
@@ -0,0 +1,103 @@
+# coding=utf-8
+"""The solution presented here goes through all parking spaces and 'places' on each of them the desired car (car which
+   on the particular space in the desired positions layout - second array). Because dictionary structure is used to
+   keep track of current cars' positions the runtime of the given solution is θ(N) where N is the number of car and the
+   space used is also θ(N)."""
+
+from Move import Move
+
+
+def create_position_map(current_positions):
+    """The method creates a dictionary which assigns each car its current position.
+
+        Args:
+            current_positions: an array of integers, indicates the current locations (layout) of cars on the parking
+
+        Returns:
+            A dictionary mapping integers to integers. Cars' numbers are keys and cars' position on the parking
+            are values.
+    """
+    position_map = {}
+    for i in range(len(current_positions)):
+        car = current_positions[i]
+        position_map[car] = i
+    return position_map
+
+
+def make_move(car, start_spot, end_spot, current_positions, position_map, moves):
+    """The method 'performs' a move, i.e. adds move to the move list and updates all position data structures.
+
+        Args:
+            car: an integer, indicates a car on a parking
+            start_spot: an integer, the space where the move 'begins'
+            end_spot: an integer, the space where the move 'ends'
+            current_positions: an array of integers, indicates the current locations (layout) of cars on the parking
+            position_map: a dictionary mapping integers to integers, cars' numbers are keys and cars' position on the
+                          parking are values
+            moves: an array of Move objects, sequence of the moves 'done so far'
+    """
+    moves.append(Move(start_spot, end_spot))
+    position_map[0] = start_spot
+    position_map[car] = end_spot
+    current_positions[start_spot] = 0
+    current_positions[end_spot] = car
+
+
+def set_single_spot(spot, current_positions, position_map, desired_positions, moves):
+    """The method sets a 'desired car' on the spot passed as an argument.
+       It uses the maximum of two moves to set the car.
+
+        Args:
+            spot: an integer, the parking space we want to set the car on
+            current_positions: an array of integers, indicates the current locations (layout) of cars on the parking
+            position_map: a dictionary mapping integers to integers, cars' numbers are keys and cars' position on the
+                          parking are values
+            desired_positions: an array of integers, indicates the desired locations (layout) of cars on the parking
+            moves: an array of Move objects, sequence of the moves 'done so far'
+    """
+    current_car = current_positions[spot]
```

Comment:
May be `current_pos` or smth like this would be better? Because it isn't a car, basically...